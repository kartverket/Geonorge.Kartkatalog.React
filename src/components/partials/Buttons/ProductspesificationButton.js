import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import style from './Buttons.scss';

export class ProductspesificationButton extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    render() {

        // TODO styling
        if (this.props.metadata.ProductSpecificationUrl) {
            let url = this.props.metadata.ProductSpecificationUrl
            let icon = <FontAwesomeIcon title="Vis produktspesifikasjon" icon={['far', 'external-link-square']} key="icon" />;
            let buttonClass = style.btn;
            let textContent = React.createElement('span', { key: "textContent" }, 'Vis produktspesifikasjon');

            let childElements = [icon, textContent];
            return React.createElement('a', { href: url, className: buttonClass }, childElements);
        } else {
            let icon = <FontAwesomeIcon title="Vis produktspesifikasjon" icon={['far', 'external-link-square']} key="icon" />
            let buttonClass = style.btn + ' disabled';
            let textContent = React.createElement('span', { key: "textContent" }, 'Vis produktspesifikasjon');
            let childElements = [icon, textContent];
            return React.createElement('span', { className: buttonClass }, childElements);
        }
    }
    
}

ProductspesificationButton.propTypes = {
    metadata: PropTypes.object.isRequired
};

export default connect(null, null)(ProductspesificationButton);
