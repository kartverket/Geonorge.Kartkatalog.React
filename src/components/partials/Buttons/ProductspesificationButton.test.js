import React from 'react';
import { shallow } from "enzyme";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { ProductspesificationButton } from './ProductspesificationButton';

function setupItemWithProductSpecificationUrl() {
    const metadata = {
        Uuid: '1234',
        ProductSpecificationUrl: 'test'
    }
    const props = {
        metadata: metadata
    }
    const wrapper = shallow(<ProductspesificationButton {...props} />)

    return {
        props,
        wrapper
    }
}

function setupItemWithoutProductSpecificationUrl() {
    const metadata = {
        Uuid: '1234',
    }
    const props = {
        metadata: metadata
    }
    const wrapper = shallow(<ProductspesificationButton {...props} />)

    return {
        props,
        wrapper
    }
}


describe('ProductsheetButton', () => {
    
    it('should render self', () => {
        const { wrapper } = setupItemWithProductSpecificationUrl()
        expect(wrapper).toMatchSnapshot();
    })

    it('Item with DistributionUrl', () => {
        const { wrapper } = setupItemWithProductSpecificationUrl()

        expect(wrapper.hasClass('btn btn-default')).toBe(true)
        expect(wrapper.prop("href")).toBe('test')
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('external-link-square')        
        expect(wrapper.find("span").first().html()).toContain('Vis produktspesifikasjon')
    })

    it('Item without DistributionUrl', () => {
        const { wrapper } = setupItemWithoutProductSpecificationUrl()
        
        expect(wrapper.hasClass('btn btn-default disabled')).toBe(true)
        expect(wrapper.prop("href")).toBeUndefined()
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('external-link-square')        
        expect(wrapper.find("span").first().html()).toContain('Vis produktspesifikasjon')
    })
})
