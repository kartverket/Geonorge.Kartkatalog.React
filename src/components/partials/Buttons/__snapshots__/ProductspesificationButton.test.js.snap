// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsheetButton should render self 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductspesificationButton
    metadata={
      Object {
        "ProductSpecificationUrl": "test",
        "Uuid": "1234",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "far",
              "external-link-square",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title="Vis produktspesifikasjon"
          transform={null}
        />,
        <span>
          Vis produktspesifikasjon
        </span>,
      ],
      "className": "btn btn-default",
      "href": "test",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "icon",
        "nodeType": "function",
        "props": Object {
          "border": false,
          "className": "",
          "fixedWidth": false,
          "flip": null,
          "icon": Array [
            "far",
            "external-link-square",
          ],
          "inverse": false,
          "listItem": false,
          "mask": null,
          "pull": null,
          "pulse": false,
          "rotation": null,
          "size": null,
          "spin": false,
          "symbol": false,
          "title": "Vis produktspesifikasjon",
          "transform": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "textContent",
        "nodeType": "host",
        "props": Object {
          "children": "Vis produktspesifikasjon",
        },
        "ref": null,
        "rendered": "Vis produktspesifikasjon",
        "type": "span",
      },
    ],
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "far",
                "external-link-square",
              ]
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title="Vis produktspesifikasjon"
            transform={null}
          />,
          <span>
            Vis produktspesifikasjon
          </span>,
        ],
        "className": "btn btn-default",
        "href": "test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "icon",
          "nodeType": "function",
          "props": Object {
            "border": false,
            "className": "",
            "fixedWidth": false,
            "flip": null,
            "icon": Array [
              "far",
              "external-link-square",
            ],
            "inverse": false,
            "listItem": false,
            "mask": null,
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "title": "Vis produktspesifikasjon",
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "textContent",
          "nodeType": "host",
          "props": Object {
            "children": "Vis produktspesifikasjon",
          },
          "ref": null,
          "rendered": "Vis produktspesifikasjon",
          "type": "span",
        },
      ],
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
